### ISSUE TITLE
Calculated OneDrive size is wrong

### ISSUE THREAD

**Issue by ferrariofilippo** [files-community/Files#10001]
### Description

When displaying OneDrive's properties, space free/used/total refers to the disk where the local folder is stored, rather than the actual size of the folder

### Steps To Reproduce

1. Open OneDrive Properties

### Steps to fix

- Fix how OneDrive used space is calculated

### Files Version

Version: 2.3.28.0 OS Version: 10.0.22000.978

### Windows Version

Windows 11 Home 21H2 22000.856

### Relevant Assets/Logs

![OneDrive](https://user-images.githubusercontent.com/102259289/190513056-a6f417c4-7fa2-4f35-9455-884026b47864.png)

**Comment by Josh-65** [comment]
Asking @yaichenbaum for approval.

**Comment by yaichenbaum** [comment]
Approved

**Comment by yaichenbaum** [comment]
@cinqmilleans do you want to look at this?

**Comment by cinqmilleans** [comment]
Ok I analyze the bug.

**Comment by cinqmilleans** [comment]
@yaichenbaum No, because the issue already exists in the preview version, before my changes. I was able to reproduce the problem, I will analyze.

**Comment by cinqmilleans** [comment]
@yaichenbaum There is a logic to this behavior. OneDrive uses a folder placed by default in the C: drive.
The sizes displayed therefore correspond to the disk that hosts the folder synchronized with OneDrive. However, the expected, intuitive behavior is to display the remote sizes. This requires code to retrieve these values. It appears that this code does not currently exist for any cloud provider.

It would be interesting to add this information in the CloudProvider class. The tooltip and the popup can display 2 sections, one for local sizes and another for remote sizes. It would be a very practical added value.

**Comment by yaichenbaum** [comment]
> No, because the issue already exists in the preview version, before my changes. I was able to reproduce the problem, I will analyze.

That's correct, this issue has been around for a while and isn't related to any of your recent changes.

> It would be interesting to add this information in the CloudProvider class. The tooltip and the popup can display 2 sections, one for local sizes and another for remote sizes. It would be a very practical added value.

I imagine the expected behavior for cloud drives is to only display the remote values and to fall back on the local values if that's the only information available.

**Comment by hecksmosis** [comment]
So this should be fixed for now by just calculating the local folder size for OneDrive?

**Comment by ferrariofilippo** [comment]
This would fix only `UsedSpace` not `TotalSpace`

**Comment by hecksmosis** [comment]
Oh true

**Comment by hecksmosis** [comment]
Is there a way to get the remote size with code?

**Comment by smarquespt** [comment]
FYI this also happens with Mega.

![image](https://user-images.githubusercontent.com/1302668/235781384-d65b9ebc-b9d3-45b1-83af-780e786b33db.png)

**Comment by yaira2** [comment]
File Explorer doesn't show any value for free space, would it be preferable to do the same for Files?

**Comment by dongle-the-gadget** [comment]
I managed to find how Explorer did it, and it's possible to implement it for OneDrive. However, from testing in a Windows 10 VM, it will only work for Windows 11, circa 23H2. Would that be an acceptable compromise for the time being?

**Comment by yaira2** [comment]
> it will only work for Windows 11, circa 23H2

Certainly 👍

**Comment by yaira2** [comment]
Initial support has been added in #16882 (thank you @dongle-the-gadget!). However, we are keeping this issue open so we can track the following:
- Enable the storage overview in the details pane for OneDrive
- Investigate the incorrect display for capacity
- Support for older versions of Windows

**Comment by yaira2** [comment]
It doesn't look like we'll be able to support this feature on older versions of Windows, however we can move forward with enabling the storage overview in the Details Pane like we do for other drives.

**Comment by yair100** [comment]
We'll continue tracking the Details Pane aspect in #17187
